index.d.ts

```diff
@@ -0,0 +1,107 @@
+declare module "react-native-markdown-view" {
+    import * as React from "react";
+    import { ViewStyle, ImageStyle, TextStyle } from "react-native";
+
+    interface MarkdownStyles {
+        [key: string]: TextStyle | ImageStyle | ViewStyle;
+        blockQuote?: TextStyle;
+        codeBlock?: TextStyle;
+        del?: TextStyle;
+        em?: TextStyle;
+        heading?: TextStyle;
+        heading1?: TextStyle;
+        heading2?: TextStyle;
+        heading3?: TextStyle;
+        heading4?: TextStyle;
+        heading5?: TextStyle;
+        heading6?: TextStyle;
+        hr?: TextStyle;
+        imageWrapper?: ViewStyle;
+        image?: ViewStyle;
+        inlineCode?: TextStyle;
+        link?: TextStyle;
+        list?: ViewStyle;
+        listItem?: ViewStyle;
+        listItemNumber?: ViewStyle;
+        listItemBullet?: ViewStyle;
+        listItemOrderedContent?: ViewStyle;
+        listItemUnorderedContent?: ViewStyle;
+        paragraph?: TextStyle;
+        strong?: TextStyle;
+        table?: ViewStyle;
+        tableHeaderCell?: ViewStyle;
+        tableHeaderCellContent?: ViewStyle;
+        tableCell?: ViewStyle;
+        tableCellOddRow?: ViewStyle;
+        tableCellEvenRow?: ViewStyle;
+        tableCellLastRow?: ViewStyle;
+        tableCellOddColumn?: ViewStyle;
+        tableCellEvenColumn?: ViewStyle;
+        tableCellLastColumn?: ViewStyle;
+        tableCellContent?: ViewStyle;
+        tableCellContentOddRow?: ViewStyle;
+        tableCellContentEvenRow?: ViewStyle;
+        tableCellContentLastRow?: ViewStyle;
+        tableCellContentOddColumn?: ViewStyle;
+        tableCellContentEvenColumn?: ViewStyle;
+        tableCellContentLastColumn?: ViewStyle;
+        u?: TextStyle;
+    }
+
+    type RegexComponents = string[]
+    type NestedParseFunction = (x: string, s: any) => any
+    type ParseState = any
+
+    type NodeKey = string
+    type OutputFunction = (node: Node, s: any) => any
+
+    interface RenderState {
+      key: string,
+      onLinkPress?: (url: string) => void
+    }
+
+    type RenderStyle = object;
+
+    interface RenderStyles {
+        [key: string]: RenderStyle
+    }
+
+    interface MarkdownRule {
+        match?: (x: string, state: RenderState, list: string[]) => RegExp | undefined | null
+        parse?: (components: RegexComponents, parse: NestedParseFunction, state: ParseState) => any,
```

Comment:
curious, why remove `ParseFunction` and `RenderFunction` types that were defined in the initial `types.js`?

Reply:
I didn't implement those types as there is no real reuse of it anywhere else. I know it was defined in types I just didn't see any value in porting it over as such.

In vscode (for instance), the type just obscures what you need to do because it presents the type as opposed to the function you need to match. Getting the type isn't that much work but this was a lot faster for the editor.